prov/rxm/src/rxm_cq.c

```diff
@@ -1979,26 +2040,51 @@ void rxm_ep_progress(struct util_ep *util_ep)
 
 void rxm_ep_progress_coll(struct util_ep *util_ep)
 {
+	struct rxm_ep *rxm_ep = container_of(util_ep, struct rxm_ep, util_ep);
+	struct util_ep *coll_ep;
+
 	rxm_ep_progress(util_ep);
 
-	ofi_coll_ep_progress(&util_ep->ep_fid);
+	if (rxm_ep->util_coll_ep) {
+		coll_ep = container_of(rxm_ep->util_coll_ep, struct util_ep,
+				       ep_fid);
+		coll_ep->progress(coll_ep);
+	}
+
+	if (rxm_ep->offload_coll_ep) {
+		coll_ep = container_of(rxm_ep->offload_coll_ep, struct util_ep,
+				       ep_fid);
+		coll_ep->progress(coll_ep);
+	}
 }
 
 static int rxm_cq_close(struct fid *fid)
 {
-	struct util_cq *util_cq;
+	struct rxm_cq *rxm_cq;
 	int ret, retv = 0;
 
-	util_cq = container_of(fid, struct util_cq, cq_fid.fid);
+	rxm_cq = container_of(fid, struct rxm_cq, util_cq.cq_fid.fid);
+
+	if (rxm_cq->offload_coll_cq)
+		fi_close((fid_t) rxm_cq->offload_coll_cq);
+
+	if (rxm_cq->util_coll_cq)
+		fi_close((fid_t) rxm_cq->util_coll_cq);
 
-	ret = ofi_cq_cleanup(util_cq);
+	ret = ofi_cq_cleanup(&rxm_cq->util_cq);
 	if (ret)
 		retv = ret;
 
-	free(util_cq);
+	free(rxm_cq);
 	return retv;
 }
 
+static int rxm_peer_cq_close(struct fid *fid)
+{
+	/* peer cq is statically allocated, nothing needs to do here */
+	return 0;
```

Comment:
I don't think anything should be calling here.

Reply:
so it is safe to set the close func to fi_no_close()?